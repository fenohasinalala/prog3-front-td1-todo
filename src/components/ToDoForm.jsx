import React from "react";
import { useState } from "react";
import { v4 as uuidv4 } from "uuid";

function ToDoForm({ onSubmitAction }) {
  const [newItem, setNewItem] = useState("");

  const handleChange = (event) => {
    setNewItem(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const nom = newItem;
    const id = uuidv4();
    const isComplet = false;
    //const id = "" + Date.now;

    if (nom.length > 0) {
      const toAdd = { id, nom, isComplet };
      onSubmitAction(toAdd);
      setNewItem("");
    }
  };

  return (
    <form action="submit" onSubmit={handleSubmit}>
      <input
        name="newTodo"
        type="text"
        value={newItem}
        placeholder="Ajouter un Item"
        onChange={handleChange}
      />
    </form>
  );
}

export default ToDoForm;
