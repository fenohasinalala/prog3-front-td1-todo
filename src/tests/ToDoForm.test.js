import { render, fireEvent, screen } from "@testing-library/react";
import ToDoForm from "../components/ToDoForm";
import renderer from "react-test-renderer";

test("It should show Item name", () => {
  const item1 = { id: 1, nom: "Do test with jest", isComplet: false };
  render(<Items itemInfo={item1} />);
  const todo = screen.getByText("Do test with jest");
  expect(todo).toBeInTheDocument();
});

test("It should show checkBox for None Completed Item", () => {
  const item1 = { id: 1, nom: "Do test with jest", isComplet: false };
  render(<Items itemInfo={item1} />);
  const ckeckbox = screen.getByRole("checkbox");
  expect(ckeckbox).toBeInTheDocument();
  expect(ckeckbox).not.toBeChecked();
});

test("It shouldn't show checkBox for completed Item", () => {
  const item1 = { id: 1, nom: "Do test with jest", isComplet: true };
  render(<Items itemInfo={item1} />);
  expect(screen.queryByRole("checkbox")).not.toBeInTheDocument();
});

test("It should use props function when click the checkbox", () => {
  const mockActionClick = jest.fn();
  const item1 = { id: 1, nom: "Do test with jest", isComplet: false };
  render(<Items itemInfo={item1} actionClick={mockActionClick} />);
  const ckeckbox = screen.getByRole("checkbox");
  fireEvent.click(ckeckbox);
  expect(mockActionClick.mock.calls.length).toBe(1);
});

it("renders one none completed Item", () => {
  const mockActionClick = jest.fn();
  const item1 = { id: 1, nom: "Do test with jest", isComplet: false };
  const component = renderer.create(
    <Items itemInfo={item1} actionClick={mockActionClick} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders one completed Item", () => {
  const mockActionClick = jest.fn();
  const item1 = { id: 1, nom: "Do test with jest", isComplet: true };
  const component = renderer.create(
    <Items itemInfo={item1} actionClick={mockActionClick} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
