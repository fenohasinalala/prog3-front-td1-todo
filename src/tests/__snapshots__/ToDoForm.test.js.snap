// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders one completed Item 1`] = `
<li
  className="item"
>
  Do test with jest
</li>
`;

exports[`renders one none completed Item 1`] = `
<li
  className="item2"
>
  Do test with jest
  <input
    onChange={[Function]}
    type="checkbox"
  />
</li>
`;
